### ISSUE TITLE
DHCP Pool Crash Checkmk Raw Edition 2.2.0b8

### ISSUE THREAD

**Issue by tzockt** [bashclub/checkmk-opnsense-agent#33]
Hi, 

ich hab mein CheckMK auf 2.2.0b8 geupdatet und erhalte bei den DHCP Pools aktuell immer crashes.
Falls weiter Infos benötigt werden, gerne im markieren :)

![image](https://github.com/bashclub/check-opnsense/assets/67980380/e4490b2d-f04c-4616-b85a-183b14509cb7)

CheckMK Exception:
```
AttributeError ('tuple' object has no attribute 'get')
```

CheckMK Traceback:
```
  File "/omd/sites/home/lib/python3/cmk/base/agent_based/checking/_checking.py", line 413, in get_aggregated_result
    consume_check_results(
  File "/omd/sites/home/lib/python3/cmk/base/api/agent_based/checking_classes.py", line 494, in consume_check_results
    for subr in subresults:
  File "/omd/sites/home/lib/python3/cmk/base/api/agent_based/register/check_plugins.py", line 93, in filtered_generator
    for element in generator(*args, **kwargs):
  File "/omd/sites/home/lib/python3/cmk/base/api/agent_based/register/check_plugins_legacy.py", line 207, in check_result_generator
    subresults = _normalize_check_function_return_value(sig_function(**kwargs))
  File "/omd/sites/home/lib/python3/cmk/base/api/agent_based/register/check_plugins_legacy.py", line 168, in _normalize_check_function_return_value
    return list(subresults)
  File "/omd/sites/home/share/check_mk/checks/isc_dhcpd", line 95, in check_isc_dhcpd
    for check_result in check_dhcp_pools_levels(
  File "/omd/sites/home/lib/python3/cmk/base/check_legacy_includes/dhcp_pools.py", line 21, in check_dhcp_pools_levels
    for new_api_object in dhcp_pools.check_dhcp_pools_levels(free, used, pending, size, params):
  File "/omd/sites/home/lib/python3/cmk/base/plugins/agent_based/utils/dhcp_pools.py", line 32, in check_dhcp_pools_levels
    if (levels := params.get(f"{category}_leases")) is not None:
```

**Comment by philidinator** [comment]
Hi,

Same Problem with CheckMK 2.2.0p2 :(

Thanks

**Comment by snafus-io** [comment]
Leider bei mir auch ;(

**Comment by obkgroove21** [comment]
Bei mir leider auch. Gints hier schon Ideen/Ansätze?

**Comment by SeeJayEmm** [comment]
Experiencing the issue here as well
CheckMK 2.2.0p4
OPNsense 23.7.1_3-amd64

**Comment by Invisi** [comment]
Going by how the other check plugins (e.g. [Windows](https://github.com/Checkmk/checkmk/blob/0e65d12ca95a307b31f2a54f9a57ea4a33cbf351/cmk/base/plugins/agent_based/win_dhcp_pools.py#L191)[0]) behave and the parameters I could find in a local crash dump[2], I'm suspecting the [isc_dhcp](https://github.com/Checkmk/checkmk/blob/0e65d12ca95a307b31f2a54f9a57ea4a33cbf351/cmk/base/legacy_checks/isc_dhcpd.py#L76) check definition[1] never got updated to the new dict-based default parameters.  
The "auto-migration-wrapper-key" magic happens [here](https://github.com/Checkmk/checkmk/blob/0e65d12ca95a307b31f2a54f9a57ea4a33cbf351/cmk/utils/check_utils.py#L80).

[0] `check_default_parameters={"free_leases": (10.0, 5.0)},`
[1] `isc_dhcpd_default_levels = (15.0, 5.0)`
[2]
```
local_vars => {..., 'params': (15.0, 5.0)}
params => Parameters({'auto-migration-wrapper-key': (15.0, 5.0)})
```

**Comment by alan7000** [comment]
Hello, is there a python developer who could patch this error? That would be nice because it's the only agent that exists and works almost perfectly.

**Comment by obkgroove21** [comment]
@thorstenspille : Kannst du hier helfen? :)

**Comment by thorstenspille** [comment]
Das ist ein Bug in checkmk, ich habe dazu einen Thread im checkmk Forum auf gemacht.

**Comment by mac-linux-free** [comment]
Ich habe auch mein Check_MK downgegraded :)

**Comment by thorstenspille** [comment]
https://forum.checkmk.com/t/check-mk-isc-dhcpd-2-2-0-crashes-if-no-leases-in-agent-output/42139

**Comment by thorstenspille** [comment]
Vorhin hab ich die Benachrichtigung von checkmk bekommen, in Version 2.2.0p15 wird es einen Fix geben.

**Comment by thorstenspille** [comment]
https://checkmk.com/werk/16323

**Comment by obkgroove21** [comment]
The problem is now fixed with the newest checkmk release 2.2.0p17:

https://forum.checkmk.com/t/release-checkmk-stable-release-2-2-0p17/43114

**Comment by alan7000** [comment]
I can confirm that the problem has been solved in version 2.2.0p17.

**Comment by JaapHoetmer** [comment]
That’s good news, and I am confirming the errors have disappeared, but I am still left with errors related to a DHCPv6 server not running on those devices where the DHCPv4 errors were shown, while DHCPv6 is not even enabled on the devices. Is that not related?

Kind regards,
Jaap

From: Flambard alan ***@***.***>
Reply to: bashclub/checkmk-opnsense-agent ***@***.***>
Date: Wednesday, 13 December 2023 at 12:21
To: bashclub/checkmk-opnsense-agent ***@***.***>
Cc: Subscribed ***@***.***>
Subject: Re: [bashclub/checkmk-opnsense-agent] DHCP Pool Crash Checkmk Raw Edition 2.2.0b8 (Issue #33)


I can confirm that the problem has been solved in version 2.2.0p17.
